package/MDAnalysis/analysis/align.py

```diff
@@ -1040,16 +1054,21 @@ def get_matching_atoms(ag1, ag2, tol_mass=0.1, strict=False):
     """
 
     if ag1.n_atoms != ag2.n_atoms:
+        if not match_atoms:
+            raise SelectionError("Mobile and reference atom selections do not "
```

Comment:
We log  the error below via `logger.error` so this one should also be logged (for consistency).